micromasters/utils.py

```diff
@@ -100,3 +100,38 @@ def get_field_names(model):
     return [
         field.name for field in model._meta.get_fields() if not field.auto_created  # pylint: disable=protected-access
     ]
+
+
+def first_matching_item(iterable, predicate):
+    """
+    Gets the first item in an iterable that matches a predicate (or None if nothing matches)
+
+    Returns:
+        Matching item or None
+    """
+    return next(filter(predicate, iterable), None)
+
+
+def is_subset_dict(dict_to_test, master_dict):
+    """
+    Checks if a dictionary is a subset of another dictionary
+
+    Args:
+        dict_to_test (dict): The subset dictionary
+        master_dict (dict): The dictionary to test against
+    Returns:
+        bool: Whether or not the first dictionary is a subset of the second
+    """
+    result = False
+    try:
+        for pkey, pvalue in dict_to_test.items():
+            if isinstance(pvalue, dict):
+                result = is_subset_dict(pvalue, master_dict[pkey])
```

Comment:
We don't have a test for the case of nested dicts, and why would we want to do this anyway? I feel like a simple for loop comparing values using `==` would be the way to go.